### PR TITLE
WIP: add benchmarking

### PR DESCRIPTION
**Pull request by madig**
I'm playing a bit with criterion, as per https://github.com/linebender/norad/issues/177.

Needs https://github.com/madig/noto-amalgamated and the glyphs2ufo output for NotoSans-MM.glyphs.

### CODE REVIEW

File: src/layer.rs [hunk 1]
```diff
@@ -41,7 +41,11 @@ impl LayerSet {
     ///
     /// The `glyph_names` argument allows norad to reuse glyph name strings,
     /// reducing memory use.
-    pub fn load(base_dir: &Path, glyph_names: &NameList) -> Result<LayerSet, Error> {
+    pub fn load(
+        base_dir: &Path,
+        glyph_names: &NameList,
+        request: &crate::DataRequest,
```
Comment (cmyr) [inline src/layer.rs:47]: I would not include `DataRequest` as part of the base API here, especially if it is only being used to thread through whether or not we load the glyph libs?

File: src/data_request.rs [hunk 2]
```diff
@@ -13,6 +13,7 @@
 pub struct DataRequest {
     pub layers: bool,
     pub lib: bool,
+    pub glyph_lib: bool,
```
Comment (cmyr) [inline src/data_request.rs:16]: Is this a useful addition, or is it only useful in the context of benchmarking?

**Review by cmyr — COMMENTED** [review]
benchmarks that involve IO are going to be especially difficult to interpret, since so much depends on hardware, read congestion, caching, etc etc. My inclination would be to just use the `load_save` example which does time those operations, which is useful enough for getting a loose sense of the impact of various changes?

If you want to do actual benchmarks, I think it is best to keep IO out of the benchmarks: load some files into memory first, and then benchmark the parsing.

I think it would also be useful to try and keep these benchmarks narrow in scope, e.g. benchmark reading just some `.glif` files, and benchmark just parsing `font_info.plist`, etc?

File: src/data_request.rs (hunk 2 above)
Comment (madig) [reply src/data_request.rs:16]: This was primarily added to test the impact of glyphlib loading on loading times. Result: it roughly doubles parsing time on my Linux and Windows machines. Ouch.

**Comment by rsheeter** [comment]
This is very old, do you plan to finish it or could it be closed?

**Pull request closed without merging**